ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/SqlLedger.scala

```diff
@@ -373,9 +373,28 @@ private class SqlLedgerFactory(ledgerDao: LedgerDao) {
                 logger.info(
                   s"Initializing ledger with ${initialLedgerEntries.length} ledger entries")
               }
-              processInitialLedgerEntries(doInit(initialId), 0, initialLedgerEntries).map { _ =>
-                initialId
-              }(DEC)
+
+              val contracts = acs.contracts
+                .map(f => Contract.fromActiveContract(f._1, f._2))
+                .to[collection.immutable.Seq]
+              val initialLedgerEnd = 0L
+              val entriesWithOffset = initialLedgerEntries.foldLeft(
+                (initialLedgerEnd, immutable.Seq.empty[(Long, LedgerEntry)]))((acc, le) => {
+                val offset = acc._1
+                val seq = acc._2
+                (offset + le.increment, seq :+ offset -> le.entry)
+              })
+
+              @SuppressWarnings(Array("org.wartremover.warts.ExplicitImplicitTypes"))
+              implicit val ec = DEC
+              for {
+                _ <- doInit(initialId)
+                _ <- ledgerDao.storeInitialState(
+                  contracts,
+                  entriesWithOffset._2,
+                  entriesWithOffset._1)
+              } yield { initialId }
```

Comment:
@francesco-da This part was written while resolving a rebase conflict. It assigns an absolute ledger offset to each ledger entry produced by the scenario loader, and gives that whole list to the DAO to persist (along with the list of active contracts produced by the scenario loader).

While writing this, I realize that because we are using the contracts produced by the scenario loader, the SQL backend will not persist contracts that have been created *and* archived within a scenario, so it will not be able to produce a valid ledger snapshot before the end of the scenario. I don't think this is an issue, just worth mentioning.